.github/workflows/deploy_to_environment_azure.yml

```diff
@@ -0,0 +1,171 @@
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Set environment variables
+        run: |
+          ENVIRONMENT=${{ github.event.inputs.environment }^^}
+
+          echo "Deploying to $ENVIRONMENT environment"
+
+          if [ "$ENVIRONMENT" == "STAGING" ]; then
+            ACA_CREDENTIALS=${{ secrets.TEST_AZURE_ACA_CREDENTIALS }}
+            ACA_CONTAINERAPP_NAME=${{ secrets.TEST_ACA_CONTAINERAPP_NAME }}
+            ACA_RESOURCE_GROUP=${{ secrets.TEST_ACA_RESOURCE_GROUP }}
+            ACR_URL=${{ secrets.TEST_ACR_URL }}
+          elif [ "$ENVIRONMENT" == "PRODUCTION"]; then
+            ACA_CREDENTIALS=${{ secrets.PRODUCTION_AZURE_ACA_CREDENTIALS }}
+            ACA_CONTAINERAPP_NAME=${{ secrets.PRODUCTION_ACA_CONTAINERAPP_NAME }}
+            ACA_RESOURCE_GROUP=${{ secrets.PRODUCTION_ACA_RESOURCE_GROUP }}
+            ACR_URL=${{ secrets.PRODUCTION_ACR_URL }}
+          else
+              echo "$ENVIRONMENT doesn't match any known options"
+              exit 1
+          fi
+
+          echo "ACA_CREDENTIALS=$ACA_CREDENTIALS" >> $GITHUB_ENV
+          echo "ACA_CONTAINERAPP_NAME=$ACA_CONTAINERAPP_NAME" >> $GITHUB_ENV
+          echo "ACA_RESOURCE_GROUP=$ACA_RESOURCE_GROUP" >> $GITHUB_ENV
+
+      - name: Azure login with ACA credentials
+        uses: azure/login@v1
+        with:
+          creds: ${{ env.ACA_CREDENTIALS }}
+
+      - name: Update Azure Container Apps Revision
+        uses: azure/CLI@v1
+        with:
+          azcliversion: 2.40.0
+          inlineScript: |
+            az config set extension.use_dynamic_install=yes_without_prompt
+            az containerapp update \
+              --name ${{ env.ACA_CONTAINERAPP_NAME }} \
+              --resource-group ${{ env.ACA_RESOURCE_GROUP }} \
+              --image ${{ env.ACR_URL }}/${{ env.DOCKER_IMAGE }}:${{ env.DOCKER_IMAGE_TAG }} \
+              --output none
+
+  # This has been disabled so that existing Github workflow tags are not overwritten
+  # create-tag:
+  #   needs: deploy-image
+  #   runs-on: ubuntu-22.04
+  #   steps:
+  #     - uses: actions/checkout@v2
+  #     - name: Create tag string
+  #       run: echo "RELEASE_TAGNAME=${{ github.event.inputs.environment }}-`date +%Y-%m-%d`.${{ github.run_number }}" >> $GITHUB_ENV
+  #     - name: Create git tag
+  #       run: |
+  #         git tag ${{ env.RELEASE_TAGNAME }}
+  #     - name: Push git tag
+  #       run: git push origin ${{ env.RELEASE_TAGNAME }}
+  #     - name: Create release
+  #       uses: actions/create-release@latest
+  #       env:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  #       with:
+  #         tag_name: ${{ env.RELEASE_TAGNAME }}
+  #         release_name: ${{ env.RELEASE_TAGNAME }}
+  #         draft: ${{ github.event.inputs.environment == 'staging' }}
+  #         prerelease: ${{ github.event.inputs.environment == 'staging' }}
+
+  cypress-tests:
+    needs: deploy-image
+    environment: staging
+    if: ${{ github.event.inputs.environment}} == staging
```

Comment:
You can see that down here the environment name check is slightly different. 
Not sure why its lower case, normally we enter it as 'Staging'